app.py

```diff
@@ -0,0 +1,470 @@
+        if len(department) != 0:
+            ids.append(or_list_search(department, departments))
+        if len(field) != 0:
+            ids.append(or_search(field, "分野", metadatas))
+        if len(yj) != 0:
+            ids.append(or_list_search(yj, yojigen))
+        if len(classtype) != 0:
+            ids.append(or_list_search(classtype, classtypes))
+        if len(language) != 0:
+            ids.append(or_search(language, "使用言語", metadatas))
+        if len(period) != 0:
+            ids.append(or_search(period, "開講年度・開講期", metadatas))
+        if len(target) != 0:
+            ids.append(or_search(target, "対象学生", metadatas))
+        if len(esubject) != 0:
+            ids.append(or_search(esubject, "E科目", metadatas))
+        if len(keyword) != 0:
+            ids.append(word_search(keyword, keywordtexts))
+        if len(professor) != 0:
+            ids.append(word_search(professor, professors))
+        if len(grade) != 0:
+            ids.append(hyoka_search(grade, int(graderatio), hyoka))
+        if len(ids) != 0:
+            store.delete(ids_union(ids))
+        
+
+        # 質問で検索するための検索候補の個数を求める
+        ID = [str(i) for i in range(len(texts))]
+        if len(ids) != 0:
+            for i in ids_union(ids):
+                ID.remove(i)
+        st.text("検索候補は" + str(len(ID)) + "個")
+
+        if len(ID) > 0:
+            # 類似度が高い順に科目を検索する
+            a = store.similarity_search(query, k=int(k))
+            st.text("該当する科目は以下の通りです。")
+            # 検索した科目を列挙する
+            for i in range(len(a)):
+                st.text(a[i].metadata["科目名"] + "   URL:" + a[i].metadata["URL"])
+            # LLMで科目の要約を生成する
+            llm = genai.GenerativeModel("gemini-pro")
+            for i in range(len(a)):
+                with st.expander(f"検索結果{i+1}"+a[i].metadata["科目名"], expanded=True):
+                    with st.spinner("シラバス要約生成中..."):
+                        result_str = ""
+                        result_str += "* 学部:  "+a[i].metadata["学部"]+"\n\n"
+                        if departments[int(a[i].metadata["ID"])] != ["なし"]:
+                            result_str += "* 学科等:  "+' '.join(departments[int(a[i].metadata["ID"])])+"\n\n"
+                        if yojigen[int(a[i].metadata["ID"])] != [""]:
+                            result_str += "* 曜時限:  "+' '.join(yojigen[int(a[i].metadata["ID"])])+"\n\n"
+                        result_str += "* 授業形態:  "+' '.join(classtypes[int(a[i].metadata["ID"])])+"\n\n"
+
+                        for k, v in a[i].metadata.items():
+                            if not k in ["科目名","学部","ID","URL"]:
+                                result_str += f"* {k}:  {v}\n\n" 
+                        
+                        sumtext = llm.generate_content(
+                            "以下の文章を日本語で箇条書きで要約を生成してください。" + fulltexts[int(a[i].metadata["ID"])]
+                        )
+                        result_str += sumtext.text
+                        st.text(result_str)

```

Comment:
st.textだとプレーンテキスト表示になってしまうので、Markdownになるように表示させてください！
```suggestion
                        st.markdown(result_str)

```